Intermittent CI failure: project.assets.json already exists

CI and PR builds of our repo fail far too often with an error like [this](https://dev.azure.com/dnceng/public/_build/results?buildId=452982&view=logs&j=6772b272-ab79-59eb-8cca-405ab5c4d2b5&t=77aa7d44-8ac6-58f9-07fa-1bd408749035&l=149).

```
##[error]stage1/bin/bootstrap/netcoreapp2.1/MSBuild/NuGet.targets(123,5): error : The file '/home/vsts/work/1/s/artifacts/obj/MSBuild/project.assets.json' already exists.
```

--- a/eng/BootStrapMSBuild.targets
+++ b/eng/BootStrapMSBuild.targets
@@ -170,12 +170,14 @@
          probably not count as a link in the cycle. -->
     <MSBuild Projects="$(MSBuildProjectFile)" Targets="Publish" BuildInParallel="$(BuildInParallel)" />
 
-    <!-- Copy all items from the publish folder to the bootstrap folder.  We might be able to just use the published
-         version as the bootstrapped version, but the extra separation here seems like it could be valuable. -->
     <ItemGroup>
+      <!-- Copy all items from the publish folder to the bootstrap folder.  We might be able to just use the published
+            version as the bootstrapped version, but the extra separation here seems like it could be valuable. -->
       <DeployedItems Include="$(PublishDir)\**\*.*" />
+
       <NuGetSdkResolverManifest Include= "$(RepoRoot)src\MSBuild\SdkResolvers\Standalone\Microsoft.Build.NuGetSdkResolver.xml" />
       <InstalledSdks Include="$(DOTNET_INSTALL_DIR)\sdk\$(DotNetCliVersion)\Sdks\**\*.*" />
+      <InstalledExtensions Include="$(DOTNET_INSTALL_DIR)\sdk\$(DotNetCliVersion)\Current\**\*.*" Exclude="$(DOTNET_INSTALL_DIR)\sdk\$(DotNetCliVersion)\Current\Microsoft.Common.props" />
     </ItemGroup>
     <Copy SourceFiles="@(DeployedItems)"
           DestinationFolder="$(BootstrapDestination)%(RecursiveDir)" />
@@ -185,6 +187,10 @@
 
     <Copy SourceFiles="@(InstalledSdks)"
           DestinationFiles="@(InstalledSdks -> '$(BootstrapDestination)Sdks\%(RecursiveDir)%(Filename)%(Extension)')" />
+
+    <Copy SourceFiles="@(InstalledExtensions)"
+          DestinationFolder="$(BootstrapDestination)Current\%(RecursiveDir)" />
+
     <Copy SourceFiles="@(_NuGetRuntimeDependencies)"
           DestinationFolder="$(BootstrapDestination)" />
   </Target>

